src/libaktualizr/package_manager/packagemanagerfake.cc

```diff
@@ -1,7 +1,20 @@
 #include "packagemanagerfake.h"
+#include "packagemanagerfactory.h"
 
 #include "utilities/fault_injection.h"
 
+extern "C" {
+__attribute__((constructor)) void packagemanagerfake_init(void);
+}
+
+void packagemanagerfake_init(void) {
+  PackageManagerFactory::registerPackageManager(
+      PACKAGE_MANAGER_NONE, [](PackageConfig pconfig, BootloaderConfig bconfig, std::shared_ptr<INvStorage> storage,
```

Comment:
It imposes additional dependencies and policies on an user package manager as it has to depend on PackageConfig, BootloaderConfig, INvStorage and accept exactly these types of params in its ctor.
A registration of instances that share a common interface would be more flexible and loosely coupled.
Having said that, I don't think that this is really important here as PackageManagerInterface is not exactly an interface and it's already depends on the aforementioned stuff and actually requires those params in its ctor.